# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: modules/libjar/nsJAR.cpp
# Commit: 6450088b6b73
# Full Hash: 6450088b6b73ffc17c79cd7097b1bfe30d00e207
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2020-09-14 21:29:41
# Regressor Bug: 1627075
# File Overlap Count: 1
# Description:
#   Bug 1656261 - Back out all recent StartupCache work r=RyanVM
#   
#   This backs out all work from bug 1627075 as well as all of its
#   descendents. There were a few conflicts when backing this out but
#   overall it was pretty clean, so I would say it's a fairly mild
# ==============================================================================

diff -r b0fabcd60430 -r 6450088b6b73 modules/libjar/nsJAR.cpp
--- a/modules/libjar/nsJAR.cpp	Mon Sep 14 17:32:40 2020 +0000
+++ b/modules/libjar/nsJAR.cpp	Mon Sep 14 17:00:53 2020 +0000
@@ -26,7 +26,7 @@
 
 // The following initialization makes a guess of 10 entries per jarfile.
 nsJAR::nsJAR()
-    : mZip(nullptr),
+    : mZip(new nsZipArchive()),
       mReleaseTime(PR_INTERVAL_NO_TIMEOUT),
       mCache(nullptr),
       mLock("nsJAR::mLock"),
@@ -87,13 +87,12 @@
 
   // The omnijar is special, it is opened early on and closed late
   // this avoids reopening it
-  RefPtr<CacheAwareZipReader> zip = mozilla::Omnijar::GetReader(zipFile);
+  RefPtr<nsZipArchive> zip = mozilla::Omnijar::GetReader(zipFile);
   if (zip) {
     mZip = zip;
     mSkipArchiveClosing = true;
     return NS_OK;
   }
-  mZip = new CacheAwareZipReader();
   return mZip->OpenArchive(zipFile);
 }
 
@@ -103,7 +102,7 @@
   if (mOpened) return NS_ERROR_FAILURE;  // Already open!
 
   nsJAR* outerJAR = static_cast<nsJAR*>(aZipReader);
-  RefPtr<CacheAwareZipReader> innerZip =
+  RefPtr<nsZipArchive> innerZip =
       mozilla::Omnijar::GetInnerReader(outerJAR->mZipFile, aZipEntry);
   if (innerZip) {
     mOpened = true;
@@ -125,12 +124,11 @@
   mOuterZipEntry.Assign(aZipEntry);
 
   RefPtr<nsZipHandle> handle;
-  rv = nsZipHandle::Init(static_cast<nsJAR*>(aZipReader)->mZip->GetZipArchive(),
+  rv = nsZipHandle::Init(static_cast<nsJAR*>(aZipReader)->mZip.get(),
                          PromiseFlatCString(aZipEntry).get(),
                          getter_AddRefs(handle));
   if (NS_FAILED(rv)) return rv;
 
-  mZip = new CacheAwareZipReader();
   return mZip->OpenArchive(handle);
 }
 
@@ -146,7 +144,6 @@
                                   getter_AddRefs(handle));
   if (NS_FAILED(rv)) return rv;
 
-  mZip = new CacheAwareZipReader();
   return mZip->OpenArchive(handle);
 }
 
@@ -168,7 +165,7 @@
   if (mSkipArchiveClosing) {
     // Reset state, but don't close the omnijar because we did not open it.
     mSkipArchiveClosing = false;
-    mZip = nullptr;
+    mZip = new nsZipArchive();
     return NS_OK;
   }
 
@@ -304,7 +301,7 @@
     return NS_ERROR_FAILURE;
   }
 
-  RefPtr<nsZipHandle> handle = mZip->GetZipArchive()->GetFD();
+  RefPtr<nsZipHandle> handle = mZip->GetFD();
   if (!handle) {
     return NS_ERROR_FAILURE;
   }